library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -452,7 +452,7 @@ public boolean isReadyToDownload(CollatedDownloadInfo collatedDownloadInfo) {
     public boolean startDownloadIfNotActive(ExecutorService executor) {
         synchronized (this) {
             boolean isActive;
-            if (mSubmittedTask == null) {
+            if (mSubmittedTask == null || mSubmittedTask.isDone()) {
```

Comment:
It didn't make sense to me to create a new thread when the existing one had completed, maybe it's needed if you're pausing/resuming though